### PR TITLE
Fix presence unmarshal with game names as numbers

### PR DESCRIPTION
**Pull request by jonas747**
When this fails decoding, it wouldn't even set the id of the guild for me, thought i was crazy at first when i saw guilds without ids, heh

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -304,16 +304,16 @@ type Game struct {
 // UnmarshalJSON unmarshals json to Game struct
 func (g *Game) UnmarshalJSON(bytes []byte) error {
 	temp := &struct {
-		Name string          `json:"name"`
+		Name json.Number     `json:"name"`
```
Comment (iopred) [inline structs.go]: This will fail on actual strings though? I think some RawMessage shenanigans is whats needed.

Comment (jonas747) [inline structs.go]: image
it won't fail

Comment (iopred) [inline structs.go]: Put "Dota 2" as one of those fields.

Comment (jonas747) [inline structs.go]: all good 
image

**Pull request merged**